kolibri_explore_plugin/assets/src/modules/coreExplore/actions.js

```diff
@@ -1,22 +1,18 @@
+import { get } from '@vueuse/core';
+import useChannels from '../../composables/useChannels';
 import { pageNameToModuleMap } from '../../constants';
 
+const { channels } = useChannels();
+
 export function resetModuleState(store, lastPageName) {
   const moduleName = pageNameToModuleMap[lastPageName];
   if (moduleName) {
     store.commit(`${moduleName}/RESET_STATE`);
   }
 }
 
-export function setAndCheckChannels(store) {
-  return store.dispatch('setChannelInfo').then(
-    channels => {
-      return channels;
-    },
-    error => {
-      store.dispatch('handleApiError', error);
-      return error;
-    }
-  );
+export function setAndCheckChannels() {
+  return get(channels);
```

Comment:
In the [upstream function by the same name](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/modules/coreLearn/actions.js#L15-L20), Kolibri is still redirecting to an error page if `!get(channels).length`. For reference, this function was originally the same as that code [from kolibri-v0.14.x](https://github.com/learningequality/kolibri/blob/release-v0.14.x/kolibri/plugins/learn/assets/src/modules/coreLearn/actions.js#L15-L28).

I'm wondering if we still want that behaviour. If we _don't_, maybe we should remove this function altogether? It, uh, isn't doing either of the things in its name :b Here's an example upstream where Kolibri is using `useChannels` directly inside a handlers.js, for an idea what the conventions are: https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/modules/recommended/handlers.js#LL10.

Reply:
Yes, I copied from upstream but I intentionally removed that `if` for now. Because our TOPICS_ROOT (aka landing page) may start the collection download if there aren't any channels. But I think that will be useful when we move the collection download to a separate page:
- `if !get(channels).length` and an starter pack was passed in config.ini: redirect to download page.
- else: redirect to CONTENT_UNAVAILABLE page.